addon/services/infinity.js

```diff
@@ -118,13 +169,142 @@ export default Service.extend({
+
+      // override `store: inject.service()` on infinity service
+      set(this, 'store', options.store);
+
+      delete options.store;
+      delete options.storeFindMethod;
+    }
+
+    set(this, 'infinityModelLoaded', get(this, 'infinityModelLoaded'));
+    set(this, 'afterInfinityModel', get(this, 'afterInfinityModel'));
+
+    // default is to start at 0, request next page and increment
+    const currentPage = options.startingPage === undefined ? 0 : options.startingPage - 1;
+    // sets first page when route is loaded
+    const firstPage = currentPage === 0 ? 1 : currentPage + 1;
+    // chunk requests by indicated perPage param
+    const perPage = options.perPage || 25;
+
+    // check if user passed in param w/ infinityModel, else check if defined on the route (for backwards compat), else default
+    const perPageParam = paramsCheck(options.perPageParam, get(this, 'perPageParam'), 'per_page');
+    const pageParam = paramsCheck(options.pageParam, get(this, 'pageParam'), 'page');
+    const totalPagesParam = paramsCheck(options.totalPagesParam, get(this, 'totalPagesParam'), 'meta.total_pages');
+
+    delete options.startingPage;
+    delete options.perPage;
+    delete options.perPageParam;
+    delete options.pageParam;
+    delete options.totalPagesParam;
+
+    let InfinityModelFactory;
+    let didPassBoundParams = !isEmpty(boundParams);
+    if (didPassBoundParams) {
+      // if pass boundParamsOrInfinityModel, send to backwards compatible mixin that sets bound params on route
+      // and subsequently looked up when user wants to load next page
+      InfinityModelFactory = InfinityModel.extend(BoundParamsMixin);
+    } else if (ExtendedInfinityModel) {
+      // if custom InfinityModel, then use as base for creating an instance
+      InfinityModelFactory = ExtendedInfinityModel;
+    } else {
+      InfinityModelFactory = InfinityModel;
+    }
+
+    let initParams = {
+      currentPage,
+      firstPage,
+      perPage,
+      perPageParam,
+      pageParam,
+      totalPagesParam,
+      _infinityModelName: modelName,
+      extraParams: options,
+      content: A()
+    };
+
+    if (didPassBoundParams) {
+      initParams._deprecatedBoundParams = boundParams;
+      initParams.route = this;
+    }
+
+    const infinityModel = InfinityModelFactory.create(initParams);
+    get(this, '_ensureCompatibility')(get(this, 'store'), get(this, '_storeFindMethod'));
+    // route specific (for backwards compat)
+    get(this, 'infinityModels').pushObject(infinityModel);
+
+    // internal service specific - TODO: hash instead of modelName
+    if (options.hashify) {
+      let _cachedCollection = get(this, '_cachedCollection');
+      if (!_cachedCollection) {
+        let _cachedCollection = hashifyInfinityCollection(get(this, '_cachedCollection'), infinityModel, modelName);
+        set(this, '_cachedCollection', _cachedCollection);
+      } else {
+        return _cachedCollection.get(modelName);
+      }
```

Comment:
This is real basic so far (just using the modelName as the hash).  If we did do it internally, I would imagine that for the routes users want to return the existing collection, we would want them to pass in something telling use to return the existing collection?  Here I have it as `options.hashify` but really could be anything else.